packages/patternfly-4/react-core/src/components/Chip/Chip.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import ChipButton from './ChipButton';
+import { Tooltip, TooltipPosition } from '../Tooltip';
+import { TimesCircleIcon } from '@patternfly/react-icons';
+import styles from '@patternfly/patternfly-next/components/Chip/chip.css';
+import GenerateId from '../../internal/GenerateId/GenerateId';
+
+class Chip extends React.Component {
+  renderOverflowChip = () => {
+    const { children, className, onClick } = this.props;
+    return (
+      <div className={css(styles.chip, styles.modifiers.overflow, className)}>
+        <ChipButton onClick={onClick} aria-label="Expand Chip">
+          <span className={css(styles.chipText)}>{children}</span>
+        </ChipButton>
+      </div>
+    );
+  }
+  renderChip = (randomId) => {
+    const { children, position, className, onClick, minTooltipChars } = this.props;
+    const ChipComponent = (
+      <div className={css(styles.chip, className)}>
+        <span className={css(styles.chipText)} id={randomId}>
+          {children}
+        </span>
+        <ChipButton onClick={onClick} aria-label="Remove" id={`remove_${randomId}`} aria-labelledby={`remove_${randomId} ${randomId}`}>
+          <TimesCircleIcon aria-hidden="true" />
+        </ChipButton>
+      </div>
+    );
+    return children.length < minTooltipChars ? (
```

Comment:
Wondering, instead of using a char number, can you check if the span is overflowing (css limits max width)? Example here: https://stackoverflow.com/questions/42012130/how-to-detect-overflow-of-react-component-without-reactdom

Reply:
After discussing, I have decided to stick with the character approach for now. checking programmatically to see if the span was overflowing did work, but the result of the calculation would have to be used to toggle the display of the tooltip on hover or on some other event. There is no way to render it conditionally based on the actual render of the element itself. Another option @jschuler mentioned was to create a flag to toggle the tooltip visibility for the chip component.